zerver/tests/test_signup.py

```diff
@@ -531,7 +535,7 @@ def test_login_bad_password(self) -> None:
     @override_settings(RATE_LIMITING_AUTHENTICATE=True)
```

Comment:
When reading this block, I got unhappy with the fact that `test_login_bad_password` didn't have a correct password check using `login_with_return`, so I just extended it to do so.  Otherwise, the test could in the future start failing for a reason other than the password being wrong, which would be a nasty way for this to bitrot.